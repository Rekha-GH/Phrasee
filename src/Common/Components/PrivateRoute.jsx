import React from 'react';
import { useSelector } from 'react-redux';
import { Redirect, Route } from 'react-router-dom';
import { getSession } from '../../Login/Store/LoginSelector';

const PrivateRoute = React.memo(({ children, ...rest }) => {
    let auth = useSelector(getSession);

    return (
        <Route
            {...rest}
            render={({ location }) =>
                auth.session ? (
                    children
                ) : (
                    <Redirect
                        to={{
                            pathname: "/login",
                            state: { from: location }
                        }}
                    />
                )
            }
        />
    );
});

PrivateRoute.displayName = 'PrivateRoute';
export default PrivateRoute;