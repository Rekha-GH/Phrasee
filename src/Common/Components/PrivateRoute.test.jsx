import React from 'react';
import toJson from 'enzyme-to-json';
import * as ReactRedux from 'react-redux';


import PrivateRoute from './PrivateRoute';
import { shallow } from '../../setupTests';

describe('PrivateRoute Tests', () => {
  const useSelectorMock = jest.spyOn(ReactRedux, 'useSelector');

  test('PrivateRoute renders correctly', () => {
    useSelectorMock.mockImplementationOnce(() => { session: 'test' }); 
    const wrapper = shallow(<PrivateRoute><div>test</div></PrivateRoute>).dive();
    
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
