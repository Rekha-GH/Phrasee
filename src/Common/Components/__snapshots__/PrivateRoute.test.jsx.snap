// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateRoute Tests PrivateRoute renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
